src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -434,4 +444,75 @@ else if(isFixed(loansPolicy)) {
       return Optional.empty();
     }
   }
+
+  public HttpResult<Loan> recall(Loan loan) {
+    final JsonObject recalls = representation
+        .getJsonObject("requestManagement", new JsonObject())
+        .getJsonObject("recalls", new JsonObject());
+
+    final HttpResult<DateTime> minimumDueDateResult =
+        getDueDate("minimumGuaranteedLoanPeriod", recalls,
+            loan.getLoanDate(), null);
+
+    // Use a java.time Clock, which allows unit tests to use a fixed clock
+    final DateTime systemDate = new DateTime(
+        ZonedDateTime.now(
+            ClockManager.getClockManager().getClock()).toInstant().toEpochMilli(),
+        DateTimeZone.UTC);
+
+    final HttpResult<DateTime> recallDueDateResult =
+        getDueDate("recallReturnInterval", recalls, systemDate, systemDate);
+
+    final List<ValidationError> errors = new ArrayList<>();
+
+    errors.addAll(combineValidationErrors(recallDueDateResult));
+    errors.addAll(combineValidationErrors(minimumDueDateResult));
+
+    if (errors.isEmpty()) {
+      final DateTime mgd = minimumDueDateResult.value();
+      final DateTime rd = recallDueDateResult.value();
+
+      final DateTime newDueDate;
+      if (mgd == null || rd.isAfter(mgd)) {
```

Comment:
Just in case. Is it applicable ? `if (rd.isAfter(mgd))` when mgd == null it means now?

I would suggest to avoid names like `rd` and `mgd` and use instead of them the following: `minimumDueDate`, `recallDueDate`

Reply:
Done. Replaced the acronyms.